src/main/java/cubicchunks/api/CubeGeneratorsRegistry.java

```diff
@@ -0,0 +1,101 @@
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.api;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.common.primitives.Ints;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+// Based on net.minecraftforge.fml.common.registry.GameRegistry.
+public class CubeGeneratorsRegistry {
+    private static Set<ICubicWorldGenerator> worldGenerators = Sets.newHashSet();
+    private static Map<ICubicWorldGenerator, Integer> worldGeneratorIndex = Maps.newHashMap();
+    private static List<ICubicWorldGenerator> sortedGeneratorList;
+
+    /**
+     * Register a cube generator - something that will change content of generated cube.
+     *
+     * @param generator           the generator
+     * @param modGenerationWeight a weight to assign to this generator. Heavy weights tend to sink to the bottom of
+     *                            list of world generators (i.e. they run later)
+     */
+    public static void registerWorldGenerator(ICubicWorldGenerator generator, int modGenerationWeight)
+    {
+        worldGenerators.add(generator);
+        worldGeneratorIndex.put(generator, modGenerationWeight);
+        if (sortedGeneratorList != null)
+        {
+            sortedGeneratorList = null;
+        }
+    }
+
+    /**
+     * Callback hook for cube gen - if your mod wishes to add extra mod related generation to the world
+     * call this
+     *
+     * @param random the cube specific {@link Random}.
+     * @param pos is a position of a block in cube with lowest world
+     *        coordinate {@link BlockPos}.
+     * @param world The minecraft {@link World} we're
+     *        generating for.
+     */
+    public static void generateWorld(Random random, BlockPos pos, World world)
+    {
+        if (sortedGeneratorList == null)
```

Comment:
You can do this at the end of registerWorldGenerator and thus avoid this if-check on every invocation of generateWorld.